edge-rtac/src/main/resources/domain/edge-rtac/features/rtac.feature

```diff
@@ -0,0 +1,156 @@
+Feature: rtac tests
+
+  Background:
+    * url baseUrl
+    * callonce login { tenant: 'diku', name: 'diku_admin', password: 'admin' }
+    * def headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*'  }
+    * def itemStatusName = 'Available'
+    * def materialTypeId = call random_uuid
+    * callonce read('classpath:domain/edge-rtac/features/util/initData.feature@PostMaterialType')
+
+  Scenario: For a non-periodical/non-serial, return holdings and item information including availability for an instance UUID
+
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@PostModeOfIssuance')
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@PostInstance')
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@PostServicePoint')
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@PostLocation')
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@PostHoldings')
+    * def createFirstItemResponse = call read('classpath:domain/edge-rtac/features/util/initData.feature@PostItem')
+    * def expectedFirstItemId = createFirstItemResponse.itemId
+    * def itemStatusName = 'Checked out'
+    * def createSecondItemResponse = call read('classpath:domain/edge-rtac/features/util/initData.feature@PostItem')
+    * def expectedSecondItemId = createSecondItemResponse.itemId
+
+    Given url edgeUrl
+    And path 'rtac/' + instanceId
+    And param apikey = apikey
+    When method GET
+    Then status 200
+    And match response.holdings.holding.length == 2
+    # api response is getting shuffled for each run, matching response by converting into an array.
+    And match [expectedFirstItemId,expectedSecondItemId] contains call expectedData response.holdings,'holdings'
+    And match ['Available','Checked out'] contains call expectedData response.holdings,'status'
+    #delete created Items
+    * call read('classpath:domain/edge-rtac/features/util/initData.feature@DeleteItems')
```

Comment:
Why is it necessary to delete the items within the test?

Reply:
Yes , it is necessary to delete because we can not create multiple materialtypes with same name in same tenant(Diku). 
So to create a new Materialtype for each run we have to delete previous materialtype and to delete materialtype it is necessary that it should not attached to any of the item.

Reply:
> Yes , it is necessary to delete because we can not create multiple materialtypes with same name in same tenant(Diku).

Rather than always try to create these records. The tests could check if a record exists with the expected name and if it does, use that record instead of creating a new one.

This means that record creation and deletion can be avoided.

Reply:
i dont think it is possible , Karate does not provide proper control statements like if-else. Only way to do it by using **EVAL** (javascript) which i dont think is good to put JS code in feature file.

Reply:
I think that is a shame, having to create different records each time and delete them in each test adds a bunch of complexity to these tests.

If we can't do it, please add comments to all of the places where we create new records and delete them (and the records that reference them) to explain why the code is this way.

Reply:
Is there anything in [this documentation](https://github.com/karatelabs/karate#conditional-logic) that could help?

Reply:
> Is there anything in [this documentation](https://github.com/karatelabs/karate#conditional-logic) that could help?

Karate provides **ternary if else** only but that will not be sufficient as we need to execute below statement 
** * callonce read('classpath:domain/edge-rtac/features/util/initData.feature@PostMaterialType')**  which is causing problem

Reply:
FYI, I did a bit of research on this, and the standard response to people asking how to do branching and conditionals in Karate seems to be "you don't." ¯\_(ツ)_/¯